lib/WOPI/Parser.php

```diff
@@ -21,40 +21,32 @@
 
 namespace OCA\Richdocuments\WOPI;
 
+use Exception;
+use OCA\Richdocuments\Service\DiscoveryService;
 use Psr\Log\LoggerInterface;
 
 class Parser {
-	/** @var DiscoveryManager */
-	private $discoveryManager;
-
-	/** @var LoggerInterface */
-	private $logger;
+	public function __construct(
+		private DiscoveryService $discoveryService,
+		private LoggerInterface $logger
+	) {
+	}
 
 	/**
-	 * @param DiscoveryManager $discoveryManager
-	 * @param LoggerInterface $logger
+	 * @throws Exception
 	 */
-	public function __construct(DiscoveryManager $discoveryManager, LoggerInterface $logger) {
-		$this->discoveryManager = $discoveryManager;
-		$this->logger = $logger;
+	public function getUrlSrcValue(string $appName): string {
+		$result = $this->getUrlSrc($appName);
+		return $result['urlsrc'];
 	}
 
 	/**
-	 * @param $mimetype
-	 * @return array
-	 * @throws \Exception
+	 * @throws Exception
 	 */
-	public function getUrlSrc($mimetype) {
-		$discovery = $this->discoveryManager->get();
+	private function getUrlSrc(string $mimetype): array {
+		$discovery = $this->discoveryService->get();
 		$this->logger->debug('WOPI::getUrlSrc discovery: {discovery}', ['discovery' => $discovery]);
-		if (\PHP_VERSION_ID < 80000) {
-			$loadEntities = libxml_disable_entity_loader(true);
-			$discoveryParsed = simplexml_load_string($discovery);
-			libxml_disable_entity_loader($loadEntities);
-		} else {
-			$discoveryParsed = simplexml_load_string($discovery);
-		}
-
+		$discoveryParsed = simplexml_load_string($discovery);
```

Comment:
I think we should require `ext-simplexml` in composer.json since we are using function `simplexml_load_string()`.

Reply:
Thanks, added 👍